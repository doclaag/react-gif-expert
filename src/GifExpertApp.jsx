import { useState } from "react";

import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Eren Jaeger"]);

  const onAddCategory = (newCategory) => {
    if (
      !categories.findIndex(
        (category) => category.toLowerCase() === newCategory.toLowerCase()
      )
    )
      return;

    // console.log(newCategory)
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1 className="text-white text-2xl font-bold mb-4">GifExpertApp</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
