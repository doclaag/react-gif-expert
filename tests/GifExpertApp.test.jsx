import { render } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Test on <GifExpertApp/>", () => {
  
    test("should render correctly", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

});
