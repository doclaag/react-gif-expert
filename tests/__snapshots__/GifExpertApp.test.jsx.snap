// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifExpertApp/> should render correctly 1`] = `
<div>
  <h1
    class="text-white text-2xl font-bold mb-4"
  >
    GifExpertApp
  </h1>
  <form
    aria-label="form"
    class="mb-4 bg-slate-800 rounded-md p-4"
  >
    <input
      class="border-2 border-gray-300 rounded-md px-2 py-1 focus:outline-none focus:border-blue-500"
      placeholder="Buscar Gifs..."
      type="text"
      value=""
    />
  </form>
  <div
    class="p-4 rounded-lg shadow-md"
  >
    <h3
      class="text-2xl font-bold text-white mb-4 text-center"
    >
      Eren Jaeger
    </h3>
    <p
      class="text-lg font-bold text-gray-800"
    >
      Loading...
    </p>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-3 xl:grid-cols-3 gap-4"
    />
  </div>
</div>
`;
